pageserver/src/tenant/timeline.rs

```diff
@@ -2476,6 +2476,30 @@ impl Timeline {
             .unwrap_or(self.conf.default_tenant_conf.lazy_slru_download)
     }
 
+    /// Checks if a get page request should get perf tracing
+    ///
+    /// The configuration priority is: tenant config override, default tenant config,
+    /// pageserver config.
+    pub(crate) fn is_get_page_request_sampled(&self) -> bool {
+        let tenant_conf = self.tenant_conf.load();
+        let ratio = tenant_conf
+            .tenant_conf
+            .sampling_ratio
+            .or(self.conf.default_tenant_conf.sampling_ratio)
+            .or(self.conf.tracing.as_ref().map(|t| t.sampling_ratio));
```

Comment:
not requesting to fix it here, but, we should really be materializing effective tenantconf to reduce this repeat pointer chasing + merging of config values _on every getpage request_

Reply:
Yeah, I highlighted this in the cover letter too. I think we can live with this for a week. I'll do a follow-up to store the materialized config.